arangod/IResearch/IResearchLink.cpp

```diff
@@ -1880,18 +1935,21 @@ Result IResearchLink::unload() {
   // this code is used by the MMFilesEngine
   // if the collection is in the process of being removed then drop it from the view
   // FIXME TODO remove once LogicalCollection::drop(...) will drop its indexes explicitly
-  if (_collection.deleted() // collection deleted
+  if (_collection.deleted()  // collection deleted
       || TRI_vocbase_col_status_e::TRI_VOC_COL_STATUS_DELETED == _collection.status()) {
     return drop();
   }
 
-  std::atomic_store(&_flushSubscription, {}); // reset together with '_asyncSelf'
-  _asyncSelf->reset(); // the data-store is being deallocated, link use is no longer valid (wait for all the view users to finish)
+  std::atomic_store(&_flushSubscription, {});  // reset together with '_asyncSelf'
+  _asyncSelf->reset();  // the data-store is being deallocated, link use is no longer valid (wait for all the view users to finish)
+
+  if (!_dataStore) {
+    return {};
+  }
+  removeStats();
```

Comment:
don't we want to class `removeStas()` regardless of `_dataStore`?

Reply:
https://github.com/arangodb/arangodb/pull/14532/files#diff-d42e2c554656ec143789da7654d09c1ad78b9e0dc2d4f2f656e856388eb8eba3R993
This code is based on this, if honestly I don't know